### DISCUSSION TITLE
ボール側、1列4行(ボールのすぐ右)、2列1行(U)･2行(J)･3行(N)が入力されない

### DISCUSSION THREAD

**Discussion by RyoIshii0213** [Yowkees/keyball#47] [Q&A] [answered]
タイトル通りなのですが、
ダイオードの通電は確認しました。
どこを確認すれば良いのでしょうか？

ボール横のキーと2列目(U列)との繋がりはわかりませんが、U列の３キーすべてが入力されないというのは･･･

**Comment by Yowkees** [comment, marked as answer]
ボールすぐ右のキーは2列4行になります。
ですので、2列目がすべて反応していない事になります。
まずProMicroの半田付けと12Pコンスルーの導通をご確認ください。

**Reply by koron** [reply]
https://twitter.com/Ryo46671803/status/1413032637980024832

> やっちゃった😭
> 赤だけ光らなかったから、LED外そうとしたらランドごと剥がれた。
> 終点？だったようで、他の6つは光ってるから、これで我慢......ダダ下がり😭😭😭

もしもコレがボール側の基板なら、配線パターンを傷つけてる可能性もありますね。

@RyoIshii0213 

もしも、このツイートに書かれた失敗がボール側の基板であれば、ハンダ付け面をその個所とPro Microまで含める形で写真撮影して共有していただけると、原因の特定となにかしらのワークアラウンドの提案ができるかもしれません。

**Reply by RyoIshii0213** [reply]
やっちゃったのは、キーを認識しないボール側ではない方で、線で繋いで点灯するようにリカバリー出来ました。
Promicroを確認してみます。

**Reply by RyoIshii0213** [reply]
半田を盛りなおして、
反応しない2列目は、Promicroの「８」と導通確認できました。
これがColの方ですよね？Rowの方がわからないのですが･･･

**Reply by RyoIshii0213** [reply]
Promicro自体の欠陥を疑い、使えているキーボードのやつでテストしてみました。
やっぱり2列目が反応なしでした。
残るは、ダイオードと基板自体でしょうか？

それぞれのダイオードの導通確認は、OKだったのですが･･･

**Reply by Yowkees** [reply]
一度ProMicroを外していただき、COL4とROW0（ProMicroのシルクでいうと8とA3）にテスターを当ててUを押してみてください。途中にダイオードがあるので、COL4(8)に赤いテスト棒を当てます。Uキーを押したときに導通しますか？

**Reply by Yowkees** [reply]
先程書きましたテストは、2列目のキーが反応していないのでおそらく導通しないと思います。

次にCOL4とUキーのスイッチソケットの右側は導通しているかご確認ください。
症状としてはここの導通が一番怪しいです。
また、COL4とVCC、COL4とGNDが繋がってしまっていないかもご確認ください。

**Reply by RyoIshii0213** [reply]
ありがとうございます。
それが、動いちゃいました。(;'∀')
また確認して、反応が無かったので、いろんなところにテスターを当て導通を確認して･･･
たまたま、正しいProMicroを刺す隣のホールとU列が導通したので、「ProMicroを刺す場所を間違えたのか？」と思い、
間違っていないことをビルドガイドで確認しつつ、試しに隣に刺してみました(ProMicroがダメになる？)。
すべて反応しなかった(当然)ので、正しい位置に刺し直して、USBを繋げて確認したところ、U列ちゃんと動きました。
必要な作業でもないですし、これといったこともしていないのですが、組み立てましたので、これで良しとします(^^ゞ
もし、また、U列が動かなかったら検証してみます。
ありがとうございました。m(__)m

**Reply by Yowkees** [reply]
無事に動くようになったようで安心しました。
Keyballの使い心地をお楽しみください。
koron様も多くのアドバイスをいただきありがとうございましたm(_ _)m
